homeassistant/components/media_player/apple_tv.py

```diff
@@ -0,0 +1,250 @@
+            return self._playing.position
+
+    @property
+    def media_position_updated_at(self):
+        """Last valid time of media position."""
+        state = self.state
+        if state == STATE_PLAYING or state == STATE_PAUSED:
+            return dt_util.utcnow()
+
+    @property
+    def entity_picture(self):
+        """Return image of the media playing."""
+        if self._playing is not None:
+            if self.artwork is not None and \
+                    self.state == STATE_PLAYING or \
+                    self.state == STATE_PAUSED:
+                url = ARTWORK_URL.format(self.entity_id, self.access_token)
+                return url
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        if self._playing is not None:
+            title = self._playing.title
+            return title if title else "No title"
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        if self._playing is not None:
+            if self.state != STATE_IDLE:
+                return SUPPORT_PAUSE | SUPPORT_PLAY | \
+                    SUPPORT_SEEK | SUPPORT_STOP | \
+                    SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+
+    def async_media_play_pause(self):
+        """Pause media on media player.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._playing is not None:
+            state = self.state
+            if state == STATE_PAUSED:
+                return self._atv.remote_control.play()
+            elif state == STATE_PLAYING:
+                return self._atv.remote_control.pause()
+
+    def async_media_play(self):
+        """Play media.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._playing is not None:
+            return self._atv.remote_control.play()
+
+    def async_media_pause(self):
+        """Pause the media player.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._playing is not None:
+            return self._atv.remote_control.pause()
+
+    def async_media_next_track(self):
+        """Send next track command.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._playing is not None:
+            return self._atv.remote_control.next()
+
+    def async_media_previous_track(self):
+        """Send previous track command.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._playing is not None:
+            return self._atv.remote_control.previous()
+
+    @asyncio.coroutine
+    def async_media_seek(self, position):
+        """Send seek command."""
+        if self._playing is not None:
+            yield from self._atv.remote_control.set_position(position)
+
+
+class ArtworkImageView(HomeAssistantView):
```

Comment:
An issue here, from what I have understood, is that the frontend uses media_entity_image to figure out if there is an image at all. Since that method returns None now, no image is shown. If I just return anything in that method, I can of course intercept the request in view and that works. Any input on how to handle this in a good way?